crates/proto/src/dnssec/rdata/sig.rs

```diff
@@ -424,17 +423,8 @@ impl BinEncodable for SIG {
     ///        by the corresponding lowercase US-ASCII letters;
     /// ```
     fn emit(&self, encoder: &mut BinEncoder<'_>) -> ProtoResult<()> {
-        let mut encoder = encoder.with_rdata_behavior(RDataEncoding::Canonical);
```

Comment:
... but here we use `encoder.with_rdata_behavior(RDataEncoding::Canonical)` instead. This seems more opinionated/high-level so we should probably stick with this?